hummingbot/market/bittrex/bittrex_auth.py

```diff
@@ -0,0 +1,63 @@
+import time
+import hmac
+import hashlib
+from typing import Dict, Any
+
+import ujson
+
+
+class BittrexAuth:
+    def __init__(self, api_key: str, secret_key: str):
+        self.api_key = api_key
+        self.secret_key = secret_key
+
+    def generate_auth_dict(
+        self,
+        http_method: str,
+        url: str,
+        params: Dict[str, Any] = None,
+        body: Dict[str, Any] = None,
+        subaccount_id: str = "",
+    ) -> Dict[str, any]:
+        """
+        Generates the url and the valid signature to authenticate with the API endpoint.
+        :param http_method: String representing the HTTP method in use ['GET', 'POST', 'DELETE'].
+        :param url: String representing the API endpoint.
+        :param params: Dictionary of url parameters to be included in the api request. USED ONLY IN SOME CASES
+        :param body: Dictionary representing the values in a request body.
+        :param subaccount_id: String value of subaccount id.
+        :return: Dictionary containing the final 'params' and its corresponding 'signature'.
+        """
+
+        # Appends params the url
+        def append_params_to_url(url: str, params: Dict[str, any] = {}) -> str:
+            if params:
+                param_str = "&".join([f"{param}={value}" for (param, value) in params.items()])
```

Comment:
It's dangerous to generate the query string this way - because it doesn't account for escape characters in urlencode format. You should use [`urllib.parse.urlencode()`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode) instead.